src/guest/start.c

```diff
@@ -513,13 +534,40 @@ static void cleanup_passthrough(void) {
 	}
 }
 
+static int check_soundcard_on_host(){
+        int ret = system("cat /proc/asound/cards | grep sof");
+        return ret;
+}
+
+static int insert_sof_tgl_snd_module() {
+	int pid;
+	int wst;
+	pid = fork();
+	if (pid == -1) {
+		fprintf(stderr, "%s: Failed to fork.", __func__);
+		return -1;
+	} else if (pid == 0) {
+		execlp("modprobe", "modprobe", "snd-sof-pci-intel-tgl", NULL);
+		return -1;
+	}else {
+		wait(&wst);
+		if (!(WIFEXITED(wst) && !WEXITSTATUS(wst))) {
+			fprintf(stderr, "Failed to load module: snd-sof-pci-intel-tgl\n");
+			return -1;
+		}
+	}
+	return 0;
+}
+
 static void cleanup(int num)
 {
 	(void)num;
 	fprintf(stderr, "Cleaning up...\n");
 	cleanup_child_proc();
 	cleanup_rpmb();
 	cleanup_passthrough();
+	if(removed_sof_tgl_snd_module)
```

Comment:
we should try to avoid using global variable. could you add the indicator in the parameter of the function like: cleanup(int num, int removed_sof_tgl_snd_module); and make the variable local in start_guest().